lager/detail/functors.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
```

Comment:
Not sure if I agree with introducing this new file, since these functors are very ad-hoc and only used as an implementation detail for the lenses `view`,`over`, etc and this is only included from `lenses.hpp`. On the other hand, it is ok maybe, I don't wanna nitpick too much.

Reply:
yeah you're right, I'm not sure why I split those from lenses.hpp